index.js

```diff
@@ -0,0 +1,89 @@
+const baseURL = "https://ghibliapi.herokuapp.com/films"
+
+document.addEventListener("DOMContentLoaded", async () => {
+    let allMoviesData = await getAllMovies();
+    let selectedMovie;
+
+    addMoviesTitlesToSelections (allMoviesData);
+
+    let SelecMenu = document.querySelector("#selectBox");
+    SelecMenu.addEventListener("change", () => {
+
+        let MoviesList = document.querySelectorAll("option");
+
+        if (MoviesList[0].selected) {
+            selectedMovie = null;
+            return; // USING THIS TO PREVENT REVIEW WHEN NO SELECTION
+        }
+
+        for (let i = 1; i < MoviesList.length; i++) {
+
+            if (MoviesList[i].selected) {
+                selectedMovie = MoviesList[i].innerText;
```

Comment:
There is a better option for handling this. You are storing the film `title` (alternatively you could store its `id`) in the `optionElement.value`. When the user picks an option the `selectElement.value` will change to have the film's `title` (or `id`). [Check this](https://www.w3schools.com/jsref/prop_select_value.asp) click `Try it yourself`. This would allow you identify the chosen movie without having to loop over the options. Currently your listener for `change` on the select has a runtime complexity of `O(n^2)` where `n` is the total number of options. Can you find out why and how to optimize it. This function could be `O(1)` constant time. One part is what I mentioned above, another has to do with a hash map or an object. Give it a try. After that talk to @MikePursuit and compare your approaches.

Reply:
Thanks for the advice, I didn't know about .value for select, really useful.
I fixed my code